src/java/htsjdk/samtools/SAMRecord.java

```diff
@@ -2144,20 +2144,15 @@ public Object clone() throws CloneNotSupportedException {
     /**
      * Returns a deep copy of the SAM record, with the following exceptions:
      *
-     *  - The header field, which shares the reference with the original record
+     *  - The header field, which shares the header reference with the original record
      *  - The file source field, which will always always be set to null in the copy
      *
      *  Note that some fields, i.e. the cigar elements, alignment blocks, and
      *  indexing bin, are not explicitly populated in the copy since they are lazily
-     *  generated on demand.
-     *
-     *  Also note that this fails:
+     *  generated on demand, which can cause this to fail:
      *
      *     original.deepCopy().equals(original)
```

Comment:
Is there a ticket to fix the remainder of the differences causing original.deepCopy().equals(original) to not be true?  It feels like a bug.